src/app/containers/ArticleTimestamp/helpers.test.js

```diff
@@ -17,4 +26,94 @@ describe('ArticleTimestamp helper functions', () => {
       expect(isValidDateTime()).toEqual(false);
     });
   });
+
+  describe('isFirstRelative', () => {
+    it(`should return true when lastPublished === firstPublished and firstPublished < 10 hours ago`, () => {
+      const oneHourAgo = timestampGenerator({ hours: 1 });
+      const firstPublished = oneHourAgo;
+      const lastPublished = oneHourAgo;
+
+      expect(isFirstRelative(lastPublished, firstPublished)).toEqual(true);
+    });
+
+    it(`should return false when lastPublished === firstPublished and firstPublished > 10 hours ago`, () => {
+      const fiveDaysAgo = timestampGenerator({ days: 5 });
+      const firstPublished = fiveDaysAgo;
+      const lastPublished = fiveDaysAgo;
+
+      expect(isFirstRelative(lastPublished, firstPublished)).toEqual(false);
+    });
+
+    it(`should return false when lastPublished !== firstPublished and firstPublished < 10 hours ago`, () => {
+      const firstPublished = timestampGenerator({ hours: 2 });
+      const lastPublished = timestampGenerator({ days: 3 });
+
+      expect(isFirstRelative(lastPublished, firstPublished)).toEqual(false);
+    });
+  });
+
+  describe('isLastRelative', () => {
+    it('should return true when lastPublished < 10 hours ago', () => {
+      const lastPublished = timestampGenerator({ hours: 8 });
+      expect(isLastRelative(lastPublished)).toEqual(true);
+    });
+
+    it('should return false when lastPublished > 10 hours ago', () => {
+      const lastPublished = timestampGenerator({ days: 1 });
+      expect(isLastRelative(lastPublished)).toEqual(false);
+    });
+  });
+
+  describe('isSameDay', () => {
+    it('should return true if both timestamps are within 24 hours of each other', () => {
+      const fortyHoursAgo = timestampGenerator({ hours: 40 });
+      const thirtySixHoursAgo = timestampGenerator({ hours: 36 });
+      expect(isSameDay(fortyHoursAgo, thirtySixHoursAgo)).toEqual(true);
+    });
+
+    it('should return false if both timestamps are not within 24 hours of each other', () => {
+      const fourHoursAgo = timestampGenerator({ hours: 4 });
+      const thirtySixHoursAgo = timestampGenerator({ hours: 36 });
+      expect(isSameDay(fourHoursAgo, thirtySixHoursAgo)).toEqual(false);
+    });
+  });
+
+  describe('isToday', () => {
+    it('should return true if timestamp is today (9 hours ago)', () => {
+      const timestamp = timestampGenerator({ minutes: 3 });
+      expect(isToday(timestamp)).toEqual(true);
+    });
+
+    it('should return false if timestamp is not today (32 hours ago)', () => {
+      const timestamp = timestampGenerator({ hours: 32 });
+      expect(isToday(timestamp)).toEqual(false);
+    });
+  });
+
+  describe('formatType', () => {
+    const dateFormats = {
+      date: 'D MMMM YYYY',
+      dateTimeTimezone: 'D MMMM YYYY, HH:mm z',
+    };
+    it(`should return default date format when firstPublished is > 10 hours ago`, () => {
```

Comment:
Hmm, this seems a little odd to me - the implementation only uses `isSameDay` and `isToday`, so why why the figure of 10 hours significant here?

This function is the most complicated and difficult one to read in the whole file IMO. So we need to make sure the expected output for every significant combination of arguments are tested, otherwise we will be unable to detect certain kinds of regression. One way of doing this is identifying every condition mentioned in the implementation, e.g.:

```
isToday(lastPublished) 
isToday(firstPublished)
lastPublished (truthiness)
firstPublished (truthiness)
isSameDay(lastPublished, firstPublished)
```
Don't forget to consider negation too (e.g. what if `firstPublished` is falsy/null?)

Reply:
makes sense